BeamAdapter_test/component/controller/InterventionalRadiologyController_test.cpp

```diff
@@ -0,0 +1,165 @@
+
+
+
+void InterventionalRadiologyController_test::testEmptyInit()
+{
+    std::string scene =
+        "<?xml version='1.0'?>                                                                  "
+        "<Node name='Root' gravity='0 -9.81 0' dt='0.01'>                                       "
+        "   <Node name='BeamModel'>                                                             "
+        "       <InterventionalRadiologyController name='DeployController' template='Rigid3d'/> "
+        "   </Node>                                                                             "
+        "</Node>                                                                                ";
+
+    EXPECT_MSG_EMIT(Error);
+    loadScene(scene);
+
+    InterventionalRadioCtrlRig3::SPtr IRCtrlPtr = m_root->get< InterventionalRadioCtrlRig3 >(sofa::core::objectmodel::BaseContext::SearchDown);
+    EXPECT_NE(IRCtrlPtr.get(), nullptr);
+    EXPECT_EQ(IRCtrlPtr->getComponentState(), ComponentState::Invalid);
+}
+
+
+void InterventionalRadiologyController_test::testDefaultInit()
+{
+    std::string scene =
+        "<?xml version='1.0'?>                                                                  "
+        "<Node name='Root' gravity='0 -9.81 0' dt='0.01'>                                       "
+        "   <Node name='BeamTopology'>                                                          "
+        "       <WireRestShape name='BeamRestShape'/>                                           "
+        "       <EdgeSetTopologyContainer name='meshLinesBeam'/>                                "
+        "       <EdgeSetTopologyModifier />                                                     "
+        "   </Node>                                                                             "
+        "   <Node name='BeamModel'>                                                             "
+        "       <MechanicalObject name='dof' template='Rigid3d' />                              "
+        "       <RegularGridTopology name='lines' nx='60' ny='1' nz='1' />                      "
+        "       <FixedConstraint name='fc' indices='0' />                                       "
+        "       <WireBeamInterpolation name='BeamInterpolation' WireRestShape='@../BeamTopology/BeamRestShape' />"
+        "       <InterventionalRadiologyController name='DeployController' template='Rigid3d' instruments='BeamInterpolation' /> "
+        "   </Node>                                                                             "
+        "</Node>                                                      ";
+
+    EXPECT_MSG_NOEMIT(Error);
+    loadScene(scene);
+
+    InterventionalRadioCtrlRig3::SPtr IRCtrlPtr = m_root->get< InterventionalRadioCtrlRig3 >(sofa::core::objectmodel::BaseContext::SearchDown);
+    EXPECT_NE(IRCtrlPtr.get(), nullptr);
+    //TODO: change component to match behavior expected by tests
+    //EXPECT_EQ(IRCtrlPtr->getComponentState(), ComponentState::Valid);
```

Comment:
How complex it is? Can it be done quickly in this PR? This is to not forget later

Reply:
It is not complex at all. In fact it has already been done in a pending PR.

The idea was to create the tests based on an old version of the plugin before all refactoring PR. Thus we make sure we didn't change the behavior with the recent PR.
And changing the componentState is part of one of the refactoring PR. I will rebase and activate the test in it.

Reply:
forget my previous answer. The component has been updated and it is now tested